src/Core/Checkout/Promotion/Aggregate/PromotionDiscount/PromotionDiscountEntity.php

```diff
@@ -258,12 +258,12 @@ public function setApplierKey(?string $applierKey): void
         $this->applierKey = $applierKey;
     }
 
-    public function getUsageKey(): string
+    public function getUsageKey(): ?string
     {
         return $this->usageKey;
     }
 
-    public function setUsageKey(string $usageKey): void
+    public function setUsageKey(?string $usageKey): void
     {
         $this->usageKey = $usageKey;
     }
```

Comment:
Instead of doing this breaking change, please just add a deprecation like the two methods above already do.